nuget/spec/dependabot/nuget/metadata_finder_spec.rb

```diff
@@ -171,6 +175,22 @@
           it { is_expected.to eq("https://github.com/dotnet/core-setup") }
         end
       end
+
+      context "that doesn't support .nuspec routes" do
+        before do
+          # registry doesn't support .nuspec route, so returns 404
+          stub_request(:get, nuget_url).to_return(status: 404)
+          # fallback begins by getting the search URL from the index
+          stub_request(:get, "https://www.myget.org/F/exceptionless/api/v3/index.json").
+            to_return(status: 200, body: fixture("nuspecs", "index.json"))
+          # next query for the package at the search URL returned
+          stub_request(:get, "https://azuresearch-usnc.nuget.org/query?prerelease=true&q=microsoft.extensions.dependencymodel&semVerLevel=2.0.0").
+            to_return(status: 200, body: fixture("nuspecs", "microsoft.extensions.depdencymodel-results.json"))
```

Comment:
❓  Should we add some examples to cover these calls turning non-200 responses?

Reply:
Good idea, I added a couple. How do those look?